### ISSUE TITLE
Objectness loss

### ISSUE THREAD

**Issue by zxj11838** [ultralytics/yolov5#1197]
## ❔Question
In the compute_loss() method, when calculating the objectness loss, only calculate the objectnessloss of the positive sample? Did the negative sample not participate in the calculation?

## Additional context
tobj[b, a, gj, gi] = (1.0 - model.gr) + model.gr * iou.detach().clamp(0).type(tobj.dtype)
 lobj += BCEobj(pi[..., 4], tobj) * balance[i]  # obj loss

**Comment by WZMIAOMIAO** [comment]
I think that objetcness loss using all sample, not only positive sample. 
The variable pi means all samples in a prediction layer.

Building the GT label corresponding to pi.
```python
tobj = torch.zeros_like(pi[..., 0], device=device)  # target obj
tobj[b, a, gj, gi] = (1.0 - model.gr) + model.gr * iou.detach().clamp(0).type(tobj.dtype)  # iou ratio
```

Calculating loss
```python
lobj += BCEobj(pi[..., 4], tobj) * balance[i]  # obj loss
```
